enumerable.rb

```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    select_array = []
+    while i < size
+      select_array << yield(self[i])
+      i += 1
+    end
+    select_array
+  end
+
+  def my_all?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    result = true
+    current = self
+    if !block_given? && pattern.nil?
+      current.my_each do |item|
+        result = false unless item
+      end
+    elsif pattern.is_a? Regexp
+      current.my_each do |item|
+        result = item =~ pattern
+      end
+    elsif pattern.is_a? Class
+      current.my_each do |item|
+        result = false unless item.is_a? pattern
+      end
+    elsif pattern
+      current.my_each do |item|
+        result = false unless item == pattern
+      end
+    elsif is_a? Array
+      current.my_each do |item|
+        result = false unless yield(item)
+      end
+    elsif is_a? Hash
+      current.my_each do |k, v|
+        result = false unless yield(k, v)
+      end
+    end
+    result
+  end
```

Comment:
- [ ] Enumerables `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex
`['saheed', 'oladele', 'suretrust'].my_all?(/d/)` should return `true`